ocaml/libs/tracing/tracing.ml

```diff
@@ -87,13 +110,39 @@ module Span = struct
     let context : SpanContext.t = {trace_id; span_id} in
     let begin_time = Unix.gettimeofday () in
     let end_time = None in
-    {context; span_kind; parent; name; begin_time; end_time; tags}
+    let status : Status.t = {status_code= Status.Unset; description= None} in
+    {context; span_kind; status; parent; name; begin_time; end_time; tags}
 
   let finish ?(tags = []) ~span () =
+    let status = span.status in
+    let tags =
+      if status.status_code = Error then
+        (Status.status_to_string status, Printexc.get_backtrace ()) :: tags
```

Comment:
This is too late, you've already called 'Status.status_to_string' which might've destroyed the backtrace.
You need to capture the backtrace as close the error as possible, i.e. ideally whatever set the status_code to Error.